samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -35,89 +35,70 @@ func NewPetApiWithBasePath(basePath string) *PetApi{
  * @param body Pet object that needs to be added to the store
  * @return void
  */
-func (a PetApi) AddPet (body Pet) (error) {
-    // verify the required parameter 'body' is set
-    if &body == nil {
-        return errors.New("Missing required parameter 'body' when calling PetApi->AddPet")
-    }
-    _sling := sling.New().Post(a.Configuration.BasePath)
+func (a PetApi) AddPet (body Pet) (ApiResponse, error) {
```

Comment:
`AddPet` does not return any `Pet` model for 2xx response: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.json#L29

ApiResponse (should be renamed to APIResponse) contains the status code, http headers and we introduce this similar to the Go Github API client (e.g. https://github.com/google/go-github/blob/master/github/repos.go#L210) just in case the developers need access to the status code and http headers.